web-animations-2/Overview.bs

```diff
@@ -304,39 +304,18 @@ follows:
         :   If |to finite timeline|,
 
         ::  1.  [=Apply any pending playback rate=] on |animation|
-            1.  If |from finite timeline| is false,
-                set |auto align start time| to true.
-            1.  Let |seek time| be zero if [=playback rate=] &ge; 0, and
-                |animation|'s <a>associated effect end</a> otherwise.
-            1.  Update the animation based on the first matching condition if any:
+            1.  set |auto align start time| to true.
+            1.  If |previous play state| is "paused" and
+                |previous progress| is resolved:
 
-                <dl class="switch">
+                1.  Set [=animation/start time=] to unresolved.
```

Comment:
Shouldn't we set the start time to unresolved when we set auto align start time to true regardless of the previous play state, as once we switch timelines, deciding to auto align the start time we don't know what the start time will be, right?

Reply:
Doing so would affect the value of anim.currentTime.  Not that currentTime is that relevant if in the process of switching timelines and needing a new start time.  My initial gut feeling was that rather than having a currentTime that can potentially become null, we should hold the last known value of currentTime until we've established a new deferred start time.  Having said that a play-pending animation will have a null value of currentTime until start time is resolved rather than 0 or effect-end that it would have if time-based.  In short, I can be convinced either way.

Reply:
A finished animation would also briefly appear to be paused if we unconditionally reset the start time if auto-aligning the start time unless we also clear the hold time.  Note that if paused we will set the hold time based on the previous progress so likely OK to clear both start and hold time.  Then we are in a consistent state with playing an animation.

Reply:
Fixed.

Made resetting the start time unconditional after setting auto-aligned start time.
Also clear hold time to avoid changing the play state to paused and added scheduling a pending play task if "running" or "finished".